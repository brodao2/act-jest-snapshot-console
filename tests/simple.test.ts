const serializer = require('../src/index');

expect.addSnapshotSerializer(serializer);

const { wrap } = serializer;

const myLog = (msg: string) => {
	console.log('Executando myFunctionTest');
	console.log(`Msg = ${msg}`);
	console.log('% % % %');
};

const myLogWithLevel = (level: 'error' | 'info' | 'warn' | 'log') => {
	console[level](`Mensagem de nÃ­vel '${level}'`);
};

const myLogWithReturn = (): string => {
	myLogWithLevel('info');
	console.log('processando...');

	return '2 + 2 = ' + (2 + 2);
};

const myLogWithNoData = (): any => {
	return undefined;
};

test('before', () => {
	expect(myLog('before')).toMatchSnapshot();
});

test('after', () => {
	expect(wrap(() => myLog('after'))).toMatchSnapshot();
});

test('error', () => {
	expect(wrap(() => myLogWithLevel('error'))).toMatchSnapshot();
});

test('info', () => {
	expect(wrap(() => myLogWithLevel('info'))).toMatchSnapshot();
});

test('log', () => {
	expect(wrap(() => myLogWithLevel('log'))).toMatchSnapshot();
});

test('warn', () => {
	expect(wrap(() => myLogWithLevel('warn'))).toMatchSnapshot();
});

test('with return', () => {
	expect(wrap(() => myLogWithReturn())).toMatchSnapshot();
});

test('with no data', () => {
	expect(wrap(() => myLogWithNoData())).toMatchSnapshot();
});
