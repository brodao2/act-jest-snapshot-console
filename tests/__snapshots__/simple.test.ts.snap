// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`after 1`] = `
(no data)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
L:Executando myFunctionTest
L:Msg = after
L:% % % %

`;

exports[`before 1`] = `undefined`;

exports[`error 1`] = `
(no data)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
E:Mensagem de nível 'error'

`;

exports[`info 1`] = `
(no data)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
I:Mensagem de nível 'info'

`;

exports[`log 1`] = `
(no data)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
L:Mensagem de nível 'log'

`;

exports[`warn 1`] = `
(no data)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
W:Mensagem de nível 'warn'

`;

exports[`with no data 1`] = `
(no data)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(no data)
`;

exports[`with return 1`] = `
2 + 2 = 4
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
I:Mensagem de nível 'info'
L:processando...

`;
